backend/internal/storage/postgres/semesters.go

```diff
@@ -0,0 +1,96 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/CamPlume1/khoury-classroom/internal/models"
+)
+
+func (db *DB) ListSemesters(ctx context.Context, orgID int64) ([]models.Semester, error) {
+	rows, err := db.connPool.Query(ctx,
+		"SELECT id, name, classroom_id, active, org_id FROM semesters WHERE org_id = $1",
+		orgID,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var semesters []models.Semester
+	for rows.Next() {
```

Comment:
Why are you manually doing this and not using pgx.CollectRows

Reply:
fixed!